packages/core/src/actions/order/createCollectionOffer.ts

```diff
@@ -0,0 +1,150 @@
+  config: Config,
+  parameters: CreateCollectionOfferParameters
+): Promise<CreateCollectionOfferResult> {
+  const {
+    starknetAccount,
+    offer: baseOrder,
+    approveInfo,
+    waitForTransaction = true
+  } = parameters;
+  const now = Math.floor(Date.now() / 1000);
+  const startDate = baseOrder.startDate || now;
+  const endDate = baseOrder.endDate || now + 30;
+  const maxEndDate = now + 60 * 60 * 24 * 30;
+  const chainId = await config.starknetProvider.getChainId();
+  const currencyAddress =
+    baseOrder.currencyAddress || config.starknetCurrencyContract;
+
+  if (startDate < Math.floor(Date.now() / 1000)) {
+    throw new Error(
+      `Invalid start date. Start date (${startDate}) cannot be in the past.`
+    );
+  }
+
+  if (endDate < startDate) {
+    throw new Error(
+      `Invalid end date. End date (${endDate}) must be after the start date (${startDate}).`
+    );
+  }
+
+  if (endDate > maxEndDate) {
+    throw new Error(
+      `End date too far in the future. End date (${endDate}) exceeds the maximum allowed (${maxEndDate}).`
+    );
+  }
+
+  if (baseOrder.startAmount === BigInt(0)) {
+    throw new Error(
+      "Invalid start amount. The start amount must be greater than zero."
+    );
+  }
+
+  if (currencyAddress !== approveInfo.currencyAddress) {
+    throw new Error("Invalid currency address. Offer and approveInfo mismatch");
+  }
+
+  const currentAllowance = await getAllowance(
+    config,
+    approveInfo.currencyAddress,
+    starknetAccount.address
+  );
+  const allowance = currentAllowance + approveInfo.amount;
+
+  const order: OrderV1 = {
+    route: RouteType.Erc20ToErc721,
+    currencyAddress:
+      baseOrder.currencyAddress ?? config.starknetCurrencyContract,
+    currencyChainId: chainId,
+    salt: 1,
+    offerer: starknetAccount.address,
+    tokenChainId: chainId,
+    tokenAddress: baseOrder.tokenAddress,
+    tokenId: new CairoOption<Uint256>(CairoOptionVariant.None),
+    quantity: cairo.uint256(1),
+    startAmount: cairo.uint256(baseOrder.startAmount),
+    endAmount: cairo.uint256(0),
+    startDate,
+    endDate,
+    brokerId: baseOrder.brokerId,
+    additionalData: []
+  };
+
+  const result = await starknetAccount.execute([
+    {
+      contractAddress: approveInfo.currencyAddress,
+      entrypoint: "approve",
+      calldata: CallData.compile({
+        spender: config.starknetExecutorContract,
+        amount: cairo.uint256(allowance)
+      })
+    },
+    {
+      contractAddress: config.starknetExecutorContract,
+      entrypoint: "create_order",
+      calldata: CallData.compile({
+        order
+      })
+    }
+  ]);
+
+  if (waitForTransaction) {
+    await config.starknetProvider.waitForTransaction(result.transaction_hash, {
+      retryInterval: 1000
+    });
+  }
+
+  const orderHash = getOrderHashFromOrderV1(order);
+
+  return {
+    orderHash,
+    transactionHash: result.transaction_hash
+  };
+}
+
+export { createCollectionOffer };
```

Comment:
Same do we need that since the only difference with a classic offer is the token ID, we should use the same function with a boolean no ?